be/src/storage/rowset/rowset.cpp

```diff
@@ -359,6 +360,9 @@ Status Rowset::get_segment_iterators(const vectorized::VectorizedSchema& schema,
     }
     seg_options.rowid_range_option = options.rowid_range_option;
     seg_options.short_key_ranges = options.short_key_ranges;
+    if (options.runtime_state != nullptr) {
+        seg_options.is_cancelled = &options.runtime_state->cancelled_ref();
```

Comment:
May the cancelled() in runtime_state be deconstructed before accessing is_cancelled?

Reply:
It may not. @mofeiatwork could it be confirmed?

Reply:
I think it won't, or it's a bug of the execution engine.